Edit README.mb

y0913 opened the pull request:
README.mbにデータベース設計を記載しました。よろしくお願いいたします。

Commit:
Edit README.mb

README.mbにデータベース設計を記載しました。よろしくお願いいたします。

Commit:
Edit README.md

Commit:
Edit README.mb

README.mdにデータベース設計を記入しました。よろしくお願いいたします。

hiroki5123 commented on README.md:
```diff
@@ -1,24 +1,49 @@
+## usersテーブル
+|column|type|option|
+|------|-------|----------------------------|
+|id|intager|null: false, foreign_key: true|
```
idは自動的に生成されるので書かなくても大丈夫です！
また、emailもdevise側で生成するので、こちらには書かなくても大丈夫です

hiroki5123 commented on README.md:
```diff
@@ -1,24 +1,49 @@
+-has_many :messages
 
-* Database initialization
+## group-usersテーブル
```
ハイフンで繋げるデータベース名はあまり見たことがないので、ハイフンではなくスネークケースにしましょう。

hiroki5123 commented on README.md:
```diff
@@ -1,24 +1,49 @@
 
-* ...
+### Association
+-belongs_to :users
```
アソシエーションの書き方がhas_manyと混ざっているような気がします。

hiroki5123 commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|column|type|option|
+|--------|-------|----------------------------|
+|id|intager|null:false, foreign_key: true|
+|user_id|integer|null: false, unique: true|
```
外部キー制約をつけましょう

hiroki5123 commented on README.md:
```diff
@@ -1,24 +1,49 @@
+|id|intager|null:false, foreign_key: true|
+|user_id|integer|null: false, unique: true|
+|group_id|integer|null: false, unique: true|
+|message|text|null: false|
```
messagesテーブルなので、messageの何かであることは明白です。このカラム名ではなんの情報が入っているのかがわかりにくいので、bodyとかtextとかに変更しましょう

Commit:
Edit README.md

README.rbを編集いたしました。
##usersテーブルにてidとemailのレコードを削除いたしました。

##group-usersテーブル→group_usersテーブルと、ハイフンをスネークケースにしました。

###Associationの書き方
belongs_to :users→belongs_to :userのように統一しました。

##messageテーブル、user_id, group_idに外部キー制約をつけました。

##messageテーブルにてcolumnのmessageをbodyに変更いたしました。

以上、よろしくお願いいたします。

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,42 @@
+
+### Assosiation
+-has_many :group-users
+-has_many :groups
```
throughオプションつけましょう

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,42 @@
+
+### Association
+-has_many :group-users
+-has_many :users
```
throughオプションつけましょう

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,42 @@
+|column|type|option|
+|--------|-------|----------------------------|
+|user_id |integer|null: false, unique: true|
+|group_id|integer|null: false, unique: true|
```
foreign_key: trueつけましょう。

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,42 @@
+|column|type|option|
+|--------|-------|----------------------------|
+|user_id|integer|null: false, unique: true, foreign_key: true|
+|group_id|integer|null: false, unique: true, foreign_key: true|
```
foreign_key: true, indexつけましょう。 unique: trueはindexつけることで必要なくなります

Commit:
Edit README.md

9行目　throughオプションをつけました。
18行目　throughオプションをつけました。
２４、２５行目foreign_key: trueをつけました。
３４行目３５行目foreign_key: trueをつけました。
よろしくお願いいたします。

Commit:
Edit README.md

README.mdを修正いたしました。
ご確認お願いいたします。

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,44 @@
 
-Things you may want to cover:
+### Assosiation
+-has_many :groups, through: :group_users
```
中間テーブルに対するアソシエーション書いてないですね

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,44 @@
 
-* Configuration
+### Association
+-has_many :users, thorough: :group_users
```
ここも

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,44 @@
+|--------|-------|----------------------------|
+|user_id|integer|null: false, foreign_key: true, index|
+|group_id|integer|null: false, foreign_key: true, index|
+|body|text|null: false|
```
imageないですか？

hzak2xx commented on README.md:
```diff
@@ -1,24 +1,44 @@
+## usersテーブル
+|column|type|option|
+|------|-------|----------------------------|
+|name|string |null: false, unique: true|
```
indexつけましょう

Commit:
Edit README.md

README.mdを編集いたしました。

ご確認お願いいたします。

Commit:
Edit README.md

##userテーブルアソシエーションに-has_many :group_usersを追記しました。
##groupテーブルにhas_many :group_usersを追加しました。

＃＃＃groupテーブルに,indexを追加しました。

##messageテーブルにimageレコードを追加しました。

よろしくお願いいたします。

ShuheiTsuji commented on README.md:
```diff
@@ -1,24 +1,49 @@
+-has_many :messages
 
-* Services (job queues, cache servers, search engines, etc.)
+## group_usersテーブル
```
groups_users にしましょうか

Commit:
Edit README.md

##group_usersをgroups_usersへ変えました。
各アソシエーションの記述もgroups_usersへ変えました。
よろしくお願いいたします。
